index.js

```diff
@@ -164,172 +161,190 @@ function buildInfo(source) {
-  }
+  return {
+    host: url.host,
+    basePath: url.path || '/',
+    //url.protocol include traling colon
+    schemes: url.protocol && [url.protocol.slice(0, -1)]
+  };
+}
+
+/*
+ * Builds a Swagger 2.0 type properites from a Swagger 1.2 type properties
+ *
+ * @param oldDataType {object} - Swagger 1.2 type object
+ *
+ * @returns {object} - Swagger 2.0 equivalent
+ */
+function buildTypeProperties(oldType) {
+  if (!oldType) { return {}; }
+
+  //TODO: handle list[<TYPE>] types from 1.1 spec
+
+  var typeMap = {
+    integer:  {type: 'integer'},
+    number:   {type: 'number'},
+    string:   {type: 'string'},
+    boolean:  {type: 'boolean'},
+    array:    {type: 'array'},
+    file:     {type: 'file'},
+    int:      {type: 'integer', format: 'int32'},
+    long:     {type: 'integer', format: 'int64'},
+    float:    {type: 'number',  format: 'float'},
+    double:   {type: 'double',  format: 'double'},
+    byte:     {type: 'string',  format: 'byte'},
+    date:     {type: 'string',  format: 'date'},
+    datetime: {type: 'string',  format: 'date-time'},
+    list:     {type: 'array'},
+    void:     {}
+  };
+
+  return typeMap[oldType.toLowerCase()] || {$ref: oldType};
 }
 
 /*
- * Process a data type object.
+ * Builds a Swagger 2.0 data type properites from a Swagger 1.2 data type properties
  *
  * @see {@link https://github.com/swagger-api/swagger-spec/blob/master/versions/
  *  1.2.md#433-data-type-fields}
  *
- * @param field {object} - A data type field
+ * @param oldDataType {object} - Swagger 1.2 data type object
  *
  * @returns {object} - Swagger 2.0 equivalent
  */
-function processDataType(field, fixRef) {
-  field = clone(field);
+function buildDataType(oldDataType) {
+  if (!oldDataType) { return; }
+  assert(typeof oldDataType === 'object');
 
-  // Checking for the existence of '#/definitions/' is related to this bug:
-  //   https://github.com/apigee-127/swagger-converter/issues/6
-  if (field.$ref && field.$ref.indexOf('#/definitions/') === -1) {
-    field.$ref = '#/definitions/' + field.$ref;
-  } else if (field.items && field.items.$ref &&
-             field.items.$ref.indexOf('#/definitions/') === -1) {
-    field.items.$ref = '#/definitions/' + field.items.$ref;
-  }
+  var result = buildTypeProperties(
+    oldDataType.type || oldDataType.dataType || oldDataType.responseClass
+  );
 
-  if (fixRef) {
-    if (field.type && primitiveTypes.indexOf(field.type) === -1) {
-      field = {$ref: '#/definitions/' + field.type};
-    }
+  //TODO: handle '0' in default
+  var defaultValue = oldDataType.default || oldDataType.defaultValue;
+  if (result.type !== 'string') {
+    defaultValue = fixNonStringValue(defaultValue);
   }
 
-  if (field.minimum) {
-    field.minimum = fixNonStringValue(field.minimum);
-  }
+  //TODO: support 'allowableValues' from 1.1 spec
 
-  if (field.maximum) {
-    field.maximum = fixNonStringValue(field.maximum);
+  var items;
+  if (result.type === 'array') {
```

Comment:
No `type` is already converted.
On line 227 I'm assigning converted `type` to result.